GitUI/UserControls/FilterToolBar.cs

```diff
@@ -85,12 +72,8 @@ private void ApplyCustomBranchFilter()
             // The user has accepted the filter
             _filterBeingChanged = false;
 
-            string filter = tscboBranchFilter.Items.Count > 0 ? tscboBranchFilter.Text : string.Empty;
-            if (filter == TranslatedStrings.NoResultsFound)
-            {
-                filter = string.Empty;
-            }
-
+            // Apply the textbox contents, no check if the (multiple) options is in tscboBranchFilter.Items (or that the list is generated)
+            string filter = tscboBranchFilter.Text == TranslatedStrings.NoResultsFound ? string.Empty : tscboBranchFilter.Text;
```

Comment:
This is the actual change in the PR
No point in checking the contents of the dropdown here